Core/Core/Login/LoginFindSchoolPresenter.swift

```diff
@@ -60,18 +64,18 @@ class LoginFindSchoolPresenter {
         loginDelegate?.openExternalURL(url)
     }
 
-    func showLoginForHost(_ host: String) {
-        let authenticationProvider = accounts.first(where: { $0.domain == host })?.authentication_provider
+    func showLoginForHost(_ host: String, authenticationProvider: String? = nil) {
+        let provider = authenticationProvider ?? accounts.first(where: { $0.domain == host })?.authentication_provider
```

Comment:
if you pressed return, rather than tapping the list item, we still want to try to use the auth provider if found.